app/models/trait_relation.rb

```diff
@@ -5,7 +5,11 @@ class TraitRelation < ActiveRecord::Base
 
   # if don't exist trait - factory, relation, create. otherwise return nil.
   def self.create_new_trait_relation(new_factory, trait)
-    return unless REDIS.sadd("trait_relations", { factory: new_factory.name, trait: trait.name }.to_json)
-    TraitRelation.create(factory_id: new_factory.id, trait_id: trait.id)
+    TraitRelation.create(factory_id: new_factory.id, trait_id: trait.id) unless same_relation_exist?(new_factory, trait)
   end
+
+  private
```

Comment:
Lint/UselessAccessModifier: Useless `private` access modifier.<br>Style/EmptyLinesAroundAccessModifier: Keep a blank line before and after `private`.